compat/win32/dirent.c

```diff
@@ -19,7 +19,7 @@ static inline void finddata2dirent(struct dirent *ent, WIN32_FIND_DATAW *fdata)
 
 	/* Set file type, based on WIN32_FIND_DATA */
 	if ((fdata->dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT)
-			&& fdata->dwReserved0 == IO_REPARSE_TAG_SYMLINK)
+			&& fdata->dwReserved0 == IO_REPARSE_TAG_MOUNT_POINT)
```

Comment:
If we agree this is the right thing to do for `IO_REPARSE_TAG_MOUNT_POINT`, wouldn't we want to keep doing the same thing for `IO_REPARSE_TAG_SYMLINK`?

Reply:
Probably; I'm happy to do both if that sounds correct, though no test changed. There is code that exists in the repo already that appears to treat both equivalently, though, not always.

Reply:
Just a drive-by observation, but yeah, I think you'd want to check for either reparse_tag value here -- but I haven't actually tried to dig into this yet.  There are several other places in the code where it looks for either type.

Reply:
Thanks! I appreicate the review. I'll add that on when I get a chance.

I would normally say that such a thing should get extracted to a helper so that everything can evenly decide what is "symlink-like", but truthfully I'm so rusty in C that I don't trust myself to make that work how people'd want 😄

Reply:
Updated the PR to allow for both.